core/class_db.cpp

```diff
@@ -275,7 +275,7 @@ bool ClassDB::is_parent_class(const StringName &p_class, const StringName &p_inh
 
 		if (inherits == p_inherits)
 			return true;
-		inherits = get_parent_class(inherits);
+		inherits = get_parent_class_nocheck(inherits);
```

Comment:
I'm not sure about this change, the "check" version was probably used for a reason. Might be needed to add `is_parent_class_nocheck` instead? But maybe let's see what @reduz thinks about it.

Reply:
Well I probably had a reason to put it in there. Maybe the check was redundant?